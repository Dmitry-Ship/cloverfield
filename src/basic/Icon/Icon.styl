.icon
  //color $color__icon
