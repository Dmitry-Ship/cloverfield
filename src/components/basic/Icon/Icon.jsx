import React, { PropTypes } from 'react';

const Icon = ({ children, name, className, onClick, size }) => {
  const style = {
    fontSize: size,
  };
  return (
    <i
      style={style}
      onClick={onClick}
      className={`${className} material-icons`}
    >
      {name}
      {children}
    </i>
  );
};

export default Icon;

Icon.defaultProps = {
  size: '24px',
  className: '',
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  children: PropTypes.any,
  onClick: PropTypes.func,
  className: PropTypes.string,
  size: PropTypes.string,
};
