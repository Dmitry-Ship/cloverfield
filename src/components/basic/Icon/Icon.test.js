import React from 'react';
import renderer from 'react-test-renderer';

import Icon from './Icon';

describe('<Icon />', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Icon name={'face'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render children when passed in', () => {
    const tree = renderer.create(
      <Icon name={'face'} >Hello World</Icon>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
