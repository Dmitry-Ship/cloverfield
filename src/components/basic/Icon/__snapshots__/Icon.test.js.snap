// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render children when passed in 1`] = `
<i
  className=" material-icons"
>
  face
  Hello World
</i>
`;

exports[`<Icon /> should render correctly 1`] = `
<i
  className=" material-icons"
>
  face
</i>
`;
